From 2d3e3aa7f7689d80016b1c5d6c6090b9ee0e1c53 Mon Sep 17 00:00:00 2001
From: Jun Miao <jun.miao@windriver.com>
Date: Thu, 19 Mar 2020 13:11:11 +0800
Subject: [PATCH] dts/t4240: remove the DMA3 device node for T4240QDS DMA
 controller

As the "T4240 Reference Manual" shown:
"The external DMA control signals are available on DMA1 and DMA2. They are not
supported by DMA3."

So, we should remove the dma3 device node in the dts. Else we would get the
below errors during doing dmatest:

modprobe dmatest run=1 iterations=42

dmatest: Started 1 threads using dma2chan0
fsl-elo-dma ffe102300.dma: chan0: Transfer Error!
fsl-elo-dma ffe102300.dma: chan0: irq: unhandled sr 0x00000080
dmatest: dma2chan0-copy0: dstbuf[0x7d8] not copied! Expected de, got 27
........................
dmatest: dma2chan7-copy0: dstbuf[0x7f7] not copied! Expected df, got 28
dmatest: dma2chan7-copy0: 5001 errors suppressed
dmatest: dma2chan7-copy0: result #42: 'data error' with src_off=0x1da1 dst_off=0x7d8 len=0x13a9 (5033)
dmatest: dma2chan7-copy0: summary 42 tests, 42 failures 76 iops 643 KB/s (0)

Signed-off-by: Jun Miao <jun.miao@windriver.com>
---
 arch/powerpc/boot/dts/fsl/t4240si-post.dtsi | 1 -
 arch/powerpc/boot/dts/fsl/t4240si-pre.dtsi  | 1 -
 2 files changed, 2 deletions(-)

diff --git a/arch/powerpc/boot/dts/fsl/t4240si-post.dtsi b/arch/powerpc/boot/dts/fsl/t4240si-post.dtsi
index 68c4eadc19e3..d9eae2d3fb65 100644
--- a/arch/powerpc/boot/dts/fsl/t4240si-post.dtsi
+++ b/arch/powerpc/boot/dts/fsl/t4240si-post.dtsi
@@ -1030,7 +1030,6 @@
 
 /include/ "elo3-dma-0.dtsi"
 /include/ "elo3-dma-1.dtsi"
-/include/ "elo3-dma-2.dtsi"
 
 /include/ "qoriq-espi-0.dtsi"
 	spi@110000 {
diff --git a/arch/powerpc/boot/dts/fsl/t4240si-pre.dtsi b/arch/powerpc/boot/dts/fsl/t4240si-pre.dtsi
index 038cf8fadee4..83b2855a5d45 100644
--- a/arch/powerpc/boot/dts/fsl/t4240si-pre.dtsi
+++ b/arch/powerpc/boot/dts/fsl/t4240si-pre.dtsi
@@ -60,7 +60,6 @@
 		usb1 = &usb1;
 		dma0 = &dma0;
 		dma1 = &dma1;
-		dma2 = &dma2;
 		sdhc = &sdhc;
 
 		fman0 = &fman0;
-- 
2.17.1

